witness/golang/cmd/witness/internal/http/server.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package http contains private implementation details for the witness server.
+package http
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/google/trillian-examples/witness/golang/api"
+	"github.com/google/trillian-examples/witness/golang/cmd/witness/internal/witness"
+	"github.com/gorilla/mux"
+)
+
+// Server is the core handler implementation of the witness.
+type Server struct {
+	w *witness.Witness
+}
+
+// NewServer creates a new server.
+func NewServer(witness *witness.Witness) *Server {
+	return &Server{
+		w: witness,
+	}
+}
+
+// update handles requests to update checkpoints.
+// It expects a JSON request consisting of bytes and a slice of slices.
```

Comment:
I guess more specifically it expects a POSTed body containing a JSON formatted `api.UpdateRequest` statement?

Reply:
Sure, always good to be precise!